src/main/java/com/accenture/application/utils/ZipUtils.java

```diff
@@ -0,0 +1,185 @@
+        out.putNextEntry(new ZipEntry(zippedName));
+        int len;
+        while ((len = in.read(buf)) > 0) {
+            out.write(buf, 0, len);
+        }
+        out.closeEntry();
+        in.close();
+        out.close();
+        return zip;
+    }
+
+    public File zipFiles(String zipOutName, Collection<String> zipEntries) throws FileNotFoundException, IOException {
+        File zip = new File(zipOutName);
+        byte[] buf = new byte[1024];
+        ZipOutputStream out = new ZipOutputStream(new FileOutputStream(zip));
+        Iterator<String> entries = zipEntries.iterator();
+        while (entries.hasNext()) {
+            String zipEntry = entries.next();
+            FileInputStream in = new FileInputStream(zipEntry);
+            Arrays.fill(buf, Byte.valueOf("0"));
+            out.putNextEntry(new ZipEntry(new File(zipEntry).getName()));
+            int len;
+            while ((len = in.read(buf)) > 0) {
+                out.write(buf, 0, len);
+            }
+            out.closeEntry();
+            in.close();
+
+        }
+
+        out.close();
+
+        return zip;
+    }
+
+    public File zipFiles(File zipOut, Collection<File> zipEntries) throws FileNotFoundException, IOException {
+        byte[] buf = new byte[1024];
+        ZipOutputStream out = new ZipOutputStream(new FileOutputStream(zipOut));
+        Iterator<File> entries = zipEntries.iterator();
+        while (entries.hasNext()) {
+            File zipEntry = entries.next();
+            FileInputStream in = new FileInputStream(zipEntry);
+            Arrays.fill(buf, Byte.valueOf("0"));
+            out.putNextEntry(new ZipEntry(zipEntry.getName()));
+            int len;
+            while ((len = in.read(buf)) > 0) {
+                out.write(buf, 0, len);
+            }
+            out.closeEntry();
+            in.close();
+
+        }
+
+        out.close();
+
+        return zipOut;
+    }
+
+    public File unzipFile(String zipName, File fileOut)
+            throws FileNotFoundException {
+
+        BufferedOutputStream out = null;
+        ZipInputStream in = null;
+
+        try {
+
+            in = new ZipInputStream(new BufferedInputStream(new FileInputStream(zipName)));
+            @SuppressWarnings("unused")
+			ZipEntry entry;
+            while ((entry = in.getNextEntry()) != null) {
+                int count;
+                byte data[] = new byte[1024];
+
+                out = new BufferedOutputStream(new FileOutputStream(fileOut.getAbsolutePath()), 1024);
+                while ((count = in.read(data, 0, 1024)) != -1) {
+                    out.write(data, 0, count);
+                }
+                out.flush();
+                IOUtils.closeQuietly(out, true);
+            }
+
+            return fileOut;
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+            return null;
+        } finally {
+            IOUtils.closeQuietly(out, true);
+            IOUtils.closeQuietly(in, true);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *in*. The resource is referenced in statements at the following lines: 93, 97, 100, 105, 114. The resource closure statement is at line: 118. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by IOUtils.closeQuietly. Either a) close *in* in a try-finally block or b) close the resource by declaring *in* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).